modules/client/src/erc20/internal/client/encoding.ts

```diff
@@ -45,11 +45,9 @@ export class ClientErc20Encoding extends ClientCore
       "initialize",
       args,
     );
-    // Strip 0x => encode in Uint8Array
-    const data = hexToBytes(strip0x(hexBytes));
     return {
       id: ERC20_PLUGIN_ID,
-      data,
+      data: hexBytes,
```

Comment:
I would suggest using the existing types (bigint, Uint8Array, etc) for standard external interoperability. People is free to reencode them however they want outside

Reply:
which has a different data type than the idiomatic one from JS regarding bytes